src/validation/func.rs

```diff
@@ -1708,11 +1485,10 @@ fn top_label(frame_stack: &StackWithLimit<BlockFrame>) -> &BlockFrame {
 		.expect("this function can't be called with empty frame stack")
 }
 
-fn require_label(
-	depth: u32,
-	frame_stack: &StackWithLimit<BlockFrame>,
-) -> Result<&BlockFrame, Error> {
-	Ok(frame_stack.get(depth as usize)?)
+fn require_label(depth: u32, frame_stack: &StackWithLimit<BlockFrame>) -> Result<&BlockFrame, Error> {
+	frame_stack
+		.get_relative_to_top(depth as usize)
+		.ok_or_else(|| Error("non-empty stack expected".into()))
```

Comment:
Since you repeat this string verbatim many times, maybe extract it to a `const`.

Reply:
Extracted to helper function.

```rust
fn underflow_err() -> Error {
    Error("non-empty stack expected".into())
}
```